app/angular/src/client/preview/decorators.ts

```diff
@@ -5,7 +5,9 @@ import { computesTemplateFromComponent } from './angular-beta/ComputesTemplateFr
 import { isComponent } from './angular-beta/utils/NgComponentAnalyzer';
 import { ICollection, NgModuleMetadata, StoryFnAngularReturnType } from './types';
 
-export const moduleMetadata = (metadata: Partial<NgModuleMetadata>) => (storyFn: () => any) => {
+export const moduleMetadata = (
+  metadata: Partial<NgModuleMetadata>
+): DecoratorFunction<StoryFnAngularReturnType> => (storyFn) => {
```

Comment:
This is a breaking change too 🙈  however I'm not sure a lot of users should be impacted by it, maybe addons authors?

Reply:
the type before :  
`(storyFn: StoryFn<StoryFnAngularReturnType>) => any`

the type after uses : 
```
export declare type DecoratorFunction<StoryFnReturnType = unknown> = (fn: StoryFn<StoryFnReturnType>, c: StoryContext) => ReturnType<StoryFn<StoryFnReturnType>>;

type ReturnType<T extends (...args: any) => any> = T extends (...args: any) => infer R ? R : any;
```

The change is on `any` (it seems to me 🤔 )  => 
`(storyFn: StoryFn<StoryFnAngularReturnType>) => StoryFnAngularReturnType`

This will specify the return type but not change behavior
Indeed if the function is badly used it is likely to require a fix of type 🤷‍♂️

Reply:
My concern was more about the 2nd argument introduced by `DecoratorFunction`: `c: StoryContext`. The previous type implied only 1 arg: `storyFn`.

Reply:
:aa: 
I forgot it. Indeed type imposes it. but it has always been present

Reply:
Yeah... I'm really not sure it is a big deal as it's mainly internal plumbing

Reply:
> Indeed type imposes it. but it has always been present

So we can consider it as a fix more than a refactoring.